lightning/src/ln/onchain_utils.rs

```diff
@@ -0,0 +1,892 @@
+				return Some(bumped_tx);
+			},
+			PackageTemplate::HolderHTLCTx { ref input } => {
+				let htlc_tx = onchain_handler.get_fully_signed_htlc_tx(&input.0, &input.1.preimage);
+				if let Some(htlc_tx) = htlc_tx {
+					// Timer set to $NEVER given we can't bump tx without anchor outputs
+					log_trace!(logger, "Going to broadcast Holder HTLC-{} claiming HTLC output {} from {}...", if input.1.preimage.is_some() { "Success" } else { "Timeout" }, input.0.vout, input.0.txid);
+					return Some(htlc_tx);
+				}
+				return None;
+			},
+			PackageTemplate::HolderCommitmentTx { ref input } => {
+				let signed_tx = onchain_handler.get_fully_signed_holder_tx(&input.1.funding_redeemscript);
+				// Timer set to $NEVER given we can't bump tx without anchor outputs
+				log_trace!(logger, "Going to broadcast Holder Transaction {} claiming funding output {} from {}...", signed_tx.txid(), input.0.vout, input.0.txid);
+				return Some(signed_tx);
+			}
+		}
+	}
+	pub(crate) fn build_malleable_justice_tx(per_commitment_point: PublicKey, per_commitment_key: SecretKey, counterparty_delayed_payment_base_key: PublicKey, counterparty_htlc_base_key: PublicKey, input_descriptor: InputDescriptors, txid: Txid, vout: u32, amount: u64, htlc: Option<HTLCOutputInCommitment>, on_counterparty_tx_csv: u16) -> Self {
+		let revk_outp = RevokedOutput {
+			per_commitment_point,
+			counterparty_delayed_payment_base_key,
+			counterparty_htlc_base_key,
+			per_commitment_key,
+			input_descriptor,
+			amount,
+			htlc,
+			on_counterparty_tx_csv,
+		};
+		let mut inputs = HashMap::with_capacity(1);
+		inputs.insert(BitcoinOutPoint { txid, vout }, revk_outp);
+		PackageTemplate::MalleableJusticeTx {
+			inputs,
+		}
+	}
+	pub(crate) fn build_counterparty_htlc_tx(per_commitment_point: PublicKey, counterparty_delayed_payment_base_key: PublicKey, counterparty_htlc_base_key: PublicKey, preimage: Option<PaymentPreimage>, htlc: HTLCOutputInCommitment, txid: Txid, vout: u32) -> Self {
+		let counterparty_outp = CounterpartyHTLCOutput {
+			per_commitment_point,
+			counterparty_delayed_payment_base_key,
+			counterparty_htlc_base_key,
+			preimage,
+			htlc
+		};
+		let mut inputs = HashMap::with_capacity(1);
+		inputs.insert(BitcoinOutPoint { txid, vout }, counterparty_outp);
+		PackageTemplate::CounterpartyHTLCTx  {
+			inputs,
+		}
+	}
+	pub(crate) fn build_holder_htlc_tx(preimage: Option<PaymentPreimage>, amount: u64, txid: Txid, vout: u32) -> Self {
+		let htlc_outp = HolderHTLCOutput {
+			preimage,
+			amount,
+		};
+		PackageTemplate::HolderHTLCTx {
+			input: (BitcoinOutPoint { txid, vout }, htlc_outp)
+		}
+	}
+	pub(crate) fn build_holder_commitment_tx(funding_redeemscript: Script, txid: Txid, vout: u32) -> Self {
+		let funding_outp = HolderFundingOutput {
+			funding_redeemscript,
+		};
+		PackageTemplate::HolderCommitmentTx {
+			input: (BitcoinOutPoint { txid, vout }, funding_outp)
+		}
+	}
+}
+
+impl Default for PackageTemplate {
```

Comment:
Wait, why is this needed?

Reply:
Dropped, and used an Option instead. Needed some base in onchaintx on which to potentially aggregate OnchainRequest.